core/txbuilder/finalize.go

```diff
@@ -45,6 +45,9 @@ func FinalizeTx(ctx context.Context, c *protocol.Chain, s Submitter, tx *legacy.
 	if err != nil {
 		return errors.Wrap(err, "tx rejected")
 	}
+	if tx.Tx.MaxTimeMs > 0 && tx.Tx.MaxTimeMs < c.TimestampMS() {
```

Comment:
Ha, was just having this discussion with @jeffomatic in Slack. Your opinion tipped the balance; removed.